src/include/sof/audio/pipeline.h

```diff
@@ -108,6 +109,8 @@ struct pipeline {
 
 	/* position update */
 	uint32_t posn_offset;		/* position update array offset*/
+
+	struct measure_context measure;
```

Comment:
Performance counters are more related to the execution plane (tasks/threads/scheduler) rather than the logic/data entities (pipelines.) Therefore it might be better to declare and allocate them inside the `struct task` and do the measurement inside the common execution related code (like scheduler) rather than within a custom task definition (like the pipeline's one) which would need to be duplicated for any other task if we want to measure their performance in future.